app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,37 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user =
+    if Setting.repository_authentication_bypass?
+      find_user_from_basic_auth
+    else
+      # Authenticate the user internally
+      perform_basic_auth
+    end
+
+    render_authorization_required unless @authenticated_user.present?
```

Comment:
:hushed: Not sure If I fully grasp your comment, but I have not touched that part in the rebase and `@authenticated_user` is set in lines 104pp. as the result of `find_user_from_basic_auth` respectively `perform_basic_auth` ?